### PR TITLE
New Base Layer to replace go-autorest

### PR DESCRIPTION
**Pull request by manicminer**
Related: https://github.com/hashicorp/pandora/pull/1309

### CODE REVIEW

**Comment by wuxu92** [comment]
Is there a schedule for this PR release? so we can use the new base layer for Automation services, I am trying to fix the broken resources that exist in azurerm. both of them are because of the autorest bug and the service team's unwilling to fix the swagger 😫

**Comment by tombuildsstuff** [comment]
The Base Layer needs to account for an error being returned via a 200 OK too, see: https://github.com/hashicorp/terraform-provider-azurerm/pull/19849#discussion_r1061670890

File: sdk/auth/github_oidc_authorizer.go [hunk 1]
```diff
@@ -8,23 +8,37 @@ import (
...
+	// ClientId is the client ID used when authenticating
+	ClientId string
+
+	// Environment is the Azure environment/cloud being targeted
+	Environment environments.Environment
+
+	// TenantId is the tenant to authenticate against
+	TenantId string
+
+	// AuxTenantIds lists additional tenants to authenticate against, currently only
```
Comment (tombuildsstuff) [inline sdk/auth/github_oidc_authorizer.go:31]:
```suggestion
	// AuxiliaryTenantIds lists additional tenants to authenticate against, currently only
```

File: sdk/auth/oidc_authorizer.go [hunk 2]
```diff
@@ -8,13 +8,24 @@ import (
...
-	Environment        environments.Environment
-	Api                environments.Api
-	TenantId           string
+	// Api describes the Azure API being used
+	Api environments.Api
+
+	// TenantId is the tenant to authenticate against
+	TenantId string
+
+	// AuxTenantIds lists additional tenants to authenticate against, currently only
```
Comment (tombuildsstuff) [inline sdk/auth/oidc_authorizer.go:20]:
```suggestion
	// AuxiliaryTenantIds lists additional tenants to authenticate against, currently only
```

**Review by tombuildsstuff — APPROVED** [review]
Left a couple of comments inline but otherwise 👍 nice work

File: sdk/environments/application_ids.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	"TargetedMessagingService":          targetedMessagingServiceAppId,
+	"TeamsServices":                     teamsServiceAppId,
+	"ThreatProtection":                  threatProtectionAppId,
+	"TrafficManager":                    trafficManagerAppId,
+	"UniversalPrint":                    universalPrintAppId,
+	"WindowsDefenderAtp":                windowsDefenderAtpAppId,
+	"WindowsVirtualDesktop":             windowsVirtualDesktopAppId,
+	"Yammer":                            yammerAppId,
+
+	// TODO: do we still need these?
```
Comment (manicminer) [inline sdk/environments/application_ids.go:112]: I'd suggest yes, at least until they are fully EOL'd everywhere?

```suggestion
	// Deprecated, remove these when APIs reach final EOL
```

**Pull request merged**